src/plugins/orangehrmPerformancePlugin/Api/PerformanceReviewAllowedActionsAPI.php

```diff
@@ -0,0 +1,164 @@
+
+    public const ACTIONABLE_STATES_MAP = [
+        WorkflowStateMachine::REVIEW_INACTIVE_SAVE => 'Save',
+        WorkflowStateMachine::REVIEW_ACTIVATE => 'Activate',
+        WorkflowStateMachine::REVIEW_IN_PROGRESS_SAVE => 'Save',
+        WorkflowStateMachine::REVIEW_COMPLETE => 'Complete'
+    ];
+
+    public const WORKFLOW_STATES_MAP = [
+        WorkflowStateMachine::REVIEW_INACTIVE_SAVE => 'SAVED',
+        WorkflowStateMachine::REVIEW_ACTIVATE => 'ACTIVATED',
+        WorkflowStateMachine::REVIEW_IN_PROGRESS_SAVE => 'IN PROGRESS',
+        WorkflowStateMachine::REVIEW_COMPLETE => 'COMPLETED'
+    ];
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $reviewId = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            self::PARAMETER_REVIEW_ID
+        );
+
+        $performanceReview = $this->getPerformanceReviewService()
+            ->getPerformanceReviewDao()
+            ->getPerformanceReviewById($reviewId);
+
+        $currentState = is_null($performanceReview) ? self::STATE_INITIAL : self::WORKFLOW_STATES_MAP[$this->getPerformanceReviewStatus($performanceReview)];
+
+        $allowedWorkflowItems = $this->getUserRoleManager()->getAllowedActions(
+            WorkflowStateMachine::FLOW_REVIEW,
+            $currentState
+        );
+
+        ksort($allowedWorkflowItems);
+
+        $actionableStates = array_map(
+            function ($workflow) {
+                $actionableState['id'] = $workflow->getAction();
+                $actionableState['label'] = self::ACTIONABLE_STATES_MAP[$workflow->getAction()];
+                return $actionableState;
+            },
+            $allowedWorkflowItems
+        );
+
+        return new EndpointCollectionResult(
+            ArrayModel::class,
+            $actionableStates,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => count($actionableStates)])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                self::PARAMETER_REVIEW_ID,
+                new Rule(Rules::POSITIVE),
+                new Rule(Rules::ENTITY_ID_EXISTS, [PerformanceReview::class]),
+                new Rule(Rules::IN_ACCESSIBLE_ENTITY_ID, [PerformanceReview::class])
+            )
+        );
+    }
+
+    /**
+     * @param PerformanceReview $performanceReview
+     * @return int
+     */
+    private function getPerformanceReviewStatus(PerformanceReview $performanceReview): int
+    {
+        if ($this->getAuthUser()->getEmpNumber() === $performanceReview->getEmployee()->getEmpNumber()) {
+            $selfReviewer = $this->getPerformanceReviewService()
+                ->getPerformanceReviewDao()
+                ->getPerformanceSelfReviewer($performanceReview);
+            return $selfReviewer->getStatus() + 1;
```

Comment:
not clear

Reply:
self status is 1 (activated), 2 (in progress), 3 (completed). Overall status is 1 (inactive), 2 (activated), 3 (in progress), 4 (completed). I have mapped label names with overall status in `ACTIONABLE_STATES_MAP`. that's why I add 1 here to get the proper status.

I'll add a comment here to make it more clear